avro-migration-helper/src/main/java/com/linkedin/avro/compatibility/AvroCompatibilityHelper.java

```diff
@@ -104,6 +109,33 @@ public static JsonDecoder newJsonDecoder(Schema schema, String input) throws IOE
     return FACTORY.newFixedField(ofType, contents);
   }
 
+  /**
+   * Create an instance of a class. If the class implements {@link SchemaConstructable},
+   * call a constructor with a {@link Schema} parameter, otherwise use a no-arg constructor.
+   * @param clazz class type to instantiate
+   * @param schema Avro schema to use for instantiating {@param clazz} if it implements {@link SchemaConstructable}
+   */
+  public static Object newInstance(Class clazz, Schema schema) {
```

Comment:
can this be constrained to IndexedRecord ?

Reply:
It can but this would be too restrictive since it would exclude `Fixed` for instance. I chose to preserve the signature of [`SpecificData.newInstance()`](https://github.com/apache/avro/blob/master/lang/java/avro/src/main/java/org/apache/avro/specific/SpecificData.java#L466)